src/content/components/IntervalMarkerOverview.js

```diff
@@ -1,44 +1,189 @@
+      }
+      this.setState({
+        hoveredItem: mouseUpItem,
+        mouseDownItem: null,
+      });
+    }
+  }
+
+  _onMouseOut() {
+    this.setState({
+      hoveredItem: null,
+    });
   }
 
   shouldComponentUpdate(nextProps, nextState) {
     return shallowCompare(this, nextProps, nextState);
   }
 
   render() {
-    const { className, rangeStart, rangeEnd, width, intervalMarkers } = this.props;
-    const itemClassName = className.split(' ')[0] + 'Item';
-    return (<ol className={className}>
-      {
-        intervalMarkers.map(({ start, dur, title, name }, i) => {
-          const pos = (start - rangeStart) / (rangeEnd - rangeStart) * width;
-          const itemWidth = dur / (rangeEnd - rangeStart) * width;
-          return <li className={itemClassName + (name ? ` ${itemClassName}Type${name}` : '')}
-                     key={i}
-                     title={title}
-                     style={{left: `${pos}px`, width: `${itemWidth}px`}}
-                     data-start={start}
-                     data-end={start + dur}
-                     onMouseDown={this._onMouseDown}
-                     onClick={this._onSelect}/>;
-        })
-      }
-    </ol>);
+    this._scheduleDraw();
+    const { className } = this.props;
+    const { mouseDownItem, hoveredItem } = this.state;
+    const title = !mouseDownItem && hoveredItem ? hoveredItem.title : null;
+    return (
+      <div className={className}>
+        <canvas className={classNames(`${className}Canvas`, 'intervalMarkerTimelineCanvas')}
+                ref={this._takeCanvasRef}
+                onMouseDown={this._onMouseDown}
+                onMouseMove={this._onMouseMove}
+                onMouseUp={this._onMouseUp}
+                onMouseOut={this._onMouseOut}
+                title={title}/>
+      </div>
+    );
+  }
+
+  _drawRoundedRect(ctx, x, y, width, height, cornerSize) {
+    // Cut out c x c -sized squares in the corners.
+    const c = Math.min(width / 2, Math.min(height / 2, cornerSize));
+    const bottom = y + height;
+    ctx.fillRect(x + c, y, width - 2 * c, c);
+    ctx.fillRect(x, y + c, width, height - 2 * c);
+    ctx.fillRect(x + c, bottom - c, width - 2 * c, c);
+  }
+
+  drawCanvas(c) {
+    const { rangeStart, rangeEnd, width, intervalMarkers, styles, overlayFills } = this.props;
+
+    const devicePixelRatio = c.ownerDocument ? c.ownerDocument.defaultView.devicePixelRatio : 1;
+    const height = c.getBoundingClientRect().height;
+    const pixelWidth = Math.round(width * devicePixelRatio);
+    const pixelHeight = Math.round(height * devicePixelRatio);
+
+    let ctx = null;
+    if (c.width !== pixelWidth || c.height !== pixelHeight) {
+      c.width = pixelWidth;
+      c.height = pixelHeight;
+      ctx = c.getContext('2d');
+    } else {
+      ctx = c.getContext('2d');
+      ctx.clearRect(0, 0, pixelWidth, pixelHeight);
+    }
+    ctx.scale(devicePixelRatio, devicePixelRatio);
```

Comment:
I believe setting the scale every draw can have some performance impacts on Gecko, at least it did when I was measuring it on the flame chart.